daemon/src/maker_inc_connections.rs

```diff
@@ -102,43 +111,40 @@ impl Actor {
         let transport_state = Arc::new(Mutex::new(transport_state));
 
         let (read, write) = stream.into_split();
-        let read = FramedRead::new(read, wire::EncryptedJsonCodec::new(transport_state.clone()))
-            .map_ok(move |msg| FromTaker { taker_id, msg })
-            .map(forward_only_ok::Message);
+        let mut read =
+            FramedRead::new(read, wire::EncryptedJsonCodec::new(transport_state.clone()))
+                .map_ok(move |msg| FromTaker { taker_id, msg });
+
+        let this = ctx.address().expect("self to be alive");
+        let taker_msg_channel = self.taker_msg_channel.clone_channel();
+        let read_fut = async move {
+            while let Ok(res) = read.try_next().await {
+                match res {
+                    Some(msg) => taker_msg_channel
+                        .send(msg)
+                        .await
+                        .expect("connection to taker to exist"),
+                    None => continue,
```

Comment:
`None` means the stream is gone, I think we should `break` here.

Reply:
Which means I think you can collapse the `while let` into `while let Ok(Some(msg))`.

Reply:
Good to know, I had a suspicion!